specs/agents/tracing-spans-destination.md

```diff
@@ -85,23 +85,16 @@ If no value is set to the `context.destination.service.resource` field, the logi
 it MUST be the following:
 
 ```groovy
-if (context.db)
-  if (context.db.instance)
-    "${subtype ?: type}/${context.db.instance}"
-  else
-    subtype ?: type
-else if (context.message)
-  if (context.message.queue?.name)
-    "${subtype ?: type}/${context.message.queue.name}"
-  else
-    subtype ?: type
-else if (context.http?.url)
-  if (context.http.url.port > 0)
-    "${context.http.url.host}:${context.http.url.port}"
-  else if (context.http.url.host)
-    context.http.url.host
+if (!span.context.service.target.name)
+  span.context.service.target.type
+else if (!span.context.service.target.type)
+  span.context.service.target.name
+else if (span.type == 'external')
+  // Special case for HTTP, gRPC, and other rpc.system spans: skip the
+  // "${service.target.type}/" prefix.
+  span.context.service.target.name
 else
-  subtype ?: type
+  "${span.context.service.target.type}/${span.context.service.target.name}"
```

Comment:
REVIEW NOTE: This section updates the logic for inferring `context.destination.service.resource` from `context.service.target.*` (and `span.type`). This is the logic I'm using in the Node.js APM agent here: https://github.com/elastic/apm-agent-nodejs/blob/a10bccfe797577f8414c957aaf1ec50ea581e8b9/lib/instrumentation/span.js#L146-L173

That link provides some more discussion on why `else if (span.type == 'external')` was used. My observation was that the exceptional case in the spec text and "otel_bridge.feature" -- where we skip the "${service.target.type}/" prefix -- is for HTTP and RPC system spans (like gRPC and Apache Dubbo).  These are exactly the set of spans that we specify should use `span.type == 'external'`.  It seems to me a much clearer check that using other attributes such as:

```
} else if (!context.db && !context.message && context.http && context.http.url) {
```

which is closer to the old logic that was used for inferring destination.service.resource. Thoughts?

Reply:
I agree, it's simpler to rely on `span.type = "external"` here, that would probably be a very good argument to enforce adherence to the spec for span types & subtypes.